### DISCUSSION TITLE
Frigate not loading when using a subpath with a reverse proxy

### DISCUSSION THREAD

**Discussion by Mario1159** [blakeblackshear/frigate#3996] [Ask A Question] [answered]
Frigate has worked well for me in the root path, but when serving the app in a subpath such as `<mydomain>/frigate` I had some problems. 
Previously I was using nginx as my reverse proxy. With just the proxy_pass specified the page didn't load, just the icon was retrieved, but when forwarding the header as followed it worked.
```
location /frigate {
  proxy_set_header Host $host;
  proxy_set_header X-Real-IP $remote_addr;
  proxy_set_header X-Ingress-Path "/frigate";
  proxy_pass http://<mydomain>:5000/;
}
```
Now I'm migrating to traefik, but I came back to the previous problem when the page stays in blank.
This is my entrypoint:
```
entrypoints:
  web:
    address: :80
    forwardedHeaders:
      insecure: true
```
And these are my docker labels:
```
labels:
  - "traefik.enable=true"
  - "traefik.http.routers.frigate.rule=(Host(`<mydomain>`) && Path(`/frigate`))"
  - "traefik.http.routers.frigate.entrypoints=web"
  - "traefik.http.services.frigate.loadbalancer.server.port=5000"
  - "traefik.http.middlewares.frigateHeaders.headers.customrequestheaders.X-Ingress-Path=/frigate"
  - "traefik.http.routers.frigate.middlewares=frigateHeaders@docker"
```

**Comment by newmindJP** [comment]
same issue here - how to fix this? is there another header missing?

**Comment by newmindJP** [comment]
I did some further investigations - whatever page I load within sub-path "/frigate/" the UI always return the default index file - e.g. I extracted the link to css file (/frigate/assets/index-9ad6681a.css) and will get the same index file as response.
in the response the sub-path is listed correctly: <body><script>window.baseUrl="/frigate/";</script>
I'm more than happy for any hint (already spent several hours on this topic)

also here in the reference I cannot see any other important headers: https://github.com/blakeblackshear/frigate-hass-addons/blob/main/frigate_proxy/rootfs/etc/nginx/includes/proxy_params.conf

**Comment by newmindJP** [comment]
found out that nginx try to access file "/opt/frigate/web/frigate/assets/index-9ad6681a.css" instead of "/opt/frigate/web/assets/index-9ad6681a.css" - seems that he add the path from the url sub-path also to the file path. Any ideas?

**Reply by vajonam** [reply]
I couldn't get the subpath to work, i have moved to subdomains as its more compatible with more apps like frigate. the instructions are on the docs page.

**Reply by newmindJP** [reply]
this is not an option for me - I could make it to work with a "dirty hack":
copied the nginx.conf locally (here is the reference: https://github.com/blakeblackshear/frigate/blob/dev/docker/main/rootfs/usr/local/nginx/conf/nginx.conf) and mapped per docker-compose
```
    volumes:
      - ./nginx.conf:/usr/local/nginx/conf/nginx.conf
```
then I added following line (~line 55) below "listen 5000;":
```
     server {
          listen 5000;
          rewrite ^/frigate(/.*)$ $1 last
```
If I read the whole file I have no idea how this could ever work just with setting header "X-Ingress-Path"

**Comment by bdoiron74** [comment, marked as answer]
I've stumbled onto this thread searching for an answer to my [question](https://github.com/blakeblackshear/frigate/issues/13006).

Seeing your "X-Ingress-Path" header option solved it for me..  and to avoid having to type the trailing slash, I'm using a redirect:

```
location = /frigate {
    return 302 /frigate/;
}

location /frigate/ {
  proxy_http_version 1.1;
  proxy_cache_bypass $http_upgrade;
  proxy_set_header Upgrade $http_upgrade;
  proxy_set_header Connection $http_connection;
  proxy_set_header Host $http_host;
  proxy_set_header X-Ingress-Path "/frigate";
  proxy_set_header Authorization "";
  proxy_pass http://frigate:5000/;
  access_log off;
}
```

**Reply by bartbutenaers** [reply]
FYI: 
For anybody interested in this topic.  My [pull request](https://github.com/blakeblackshear/frigate/pull/17030#issuecomment-2727659313) (which was inspired by the above manual fix from @bdoiron74) has been merged, so in the near future this feature will be available in Frigate out of the box.

**Reply by neagix** [reply]
@bartbutenaers the documentation changes from your PR do not seem to be live anywhere, not even on `dev` branch.

**Reply by philios33** [reply]
My problem was I was using a backend variable in my proxy_pass which does not work, even if the variable contains a trailing slash.  The /frigate/ part was not being removed from the proxied path and I was getting the same default response page for all resource requests.  Your proxy_pass value must have a trailing / for the path to get removed properly.  It is a weird nginx ism.

**Comment by graealex** [comment]
Just a heads-up - even if Frigate overall loads, the go2rtc live streams will not properly work with subfolders. The issue is now exacerbated with Chrome recently switching to HTTPS only.

**Reply by bdoiron74** [reply]
Live streams seem to be working for me with frigate 0.15-1 (6cb5cfb) subfolder'ed to http://myhost/frigate/. I don't get WebRTC, but I do see the live streams playing through:
```
ws://myhost/frigate/live/mse/api/ws?src=CAM-A20031
```

**Reply by bdoiron74** [reply]
oh - should mention I am still using my proxy settings [here](https://github.com/blakeblackshear/frigate/discussions/3996#discussioncomment-10318509) - I haven't tried the new FRIGATE_BASE_PATH variable

**Comment by kravemir** [comment]
Sharing my working docker compose setup for frigate, traefik and HTTPS(acme-dns).

It has also the redirect of `/frigate` to `/frigate/`.

```docker-compose
services:
  traefik:
    image: traefik:v3.7
    restart: unless-stopped
    ports:
      - "80:80"
      - "443:443"
    environment:
      - ACME_DNS_STORAGE_PATH=/acme/credentials.json
      - ACME_DNS_API_BASE=https://auth.acme-dns.io
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock:ro
      - /storage/ssddata/traefik/acme:/acme
    command:
      - "--api.dashboard=true"
      - "--api.insecure=false"
      - "--providers.docker=true"
      - "--providers.docker.exposedByDefault=false"
      - "--entryPoints.web.address=:80"
      - "--entryPoints.websecure.address=:443"
      - "--entryPoints.web.http.redirections.entryPoint.to=websecure"
      - "--entryPoints.web.http.redirections.entryPoint.scheme=https"
      - "--certificatesResolvers.acme-dns.acme.email=my.email@gmail.com"
      - "--certificatesResolvers.acme-dns.acme.storage=/acme/acme.json"
      - "--certificatesResolvers.acme-dns.acme.dnsChallenge.provider=acmedns"
      - "--certificatesResolvers.acme-dns.acme.dnsChallenge.resolvers=1.1.1.1:53,8.8.8.8:53"
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.dashboard.entrypoints=websecure"
      - "traefik.http.routers.dashboard.rule=Host(`nas.at.my.domain)"
      - "traefik.http.routers.dashboard.tls=true"
      - "traefik.http.routers.dashboard.tls.certresolver=acme-dns"
      - "traefik.http.routers.dashboard.service=api@internal"
    networks:
      - proxy

  frigate:
    image: ghcr.io/blakeblackshear/frigate:0.17.2
    restart: unless-stopped
    shm_size: "512mb"
    devices:
      - /dev/apex_0:/dev/apex_0
      - /dev/dri/renderD128:/dev/dri/renderD128
    volumes:
      - /etc/localtime:/etc/localtime:ro
      - /storage/ssddata/frigate/config:/config
      - /storage/hdddata/frigate/storage:/media/frigate
      - type: tmpfs
        target: /tmp/cache
        tmpfs:
          size: 1000000000
    networks:
      - proxy
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.frigate.rule=Host(`nas.at.my.domain`) && (Path(`/frigate`) || PathPrefix(`/frigate/`))"
      - "traefik.http.routers.frigate.entrypoints=websecure"
      - "traefik.http.routers.frigate.tls=true"
      - "traefik.http.routers.frigate.tls.certresolver=acme-dns"
      - "traefik.http.middlewares.frigate-slash-redirect.redirectregex.regex=(^https://[^/]+/frigate)$$"
      - "traefik.http.middlewares.frigate-slash-redirect.redirectregex.replacement=$$1/"
      - "traefik.http.middlewares.frigate-strip.stripprefix.prefixes=/frigate"
      - "traefik.http.middlewares.frigate-headers.headers.customrequestheaders.X-Ingress-Path=/frigate"
      - "traefik.http.routers.frigate.middlewares=frigate-slash-redirect,frigate-strip,frigate-headers"
      - "traefik.http.services.frigate.loadbalancer.server.port=8971"
    ports:
      - "8971:8971"
      # - "5000:5000" # Internal unauthenticated access. Expose carefully.
      # - "8554:8554" # RTSP feeds (unauthenticated by default)
      - "8555:8555/tcp" # WebRTC over tcp
      - "8555:8555/udp" # WebRTC over udp
      # - "1984:1984" # go2rtc test
    cap_add:
      - CAP_PERFMON

networks:
  proxy:
    name: proxy
```
